### PR TITLE
adapt PlotHandler to use (x,y)-Tupels

### PR DESCRIPTION
**Pull request by kohlros**
PlotWorkers now save the data in {xValue,yValue}-Objects to allow for an x-y-plotting block.

### CODE REVIEW

File: static/scripts/PlotHandler.js [hunk 1]
```diff
@@ -54,17 +51,25 @@ class PlotWorker {
...
+                borderColor: randomColor, 
+                data: [{ x: data.xValue, y: data.yValue }],
             }
             this.plotData.set(datasetName, dataset);
         } else {
-            dataset.data.push(data.yValue);
-            if (dataset.data.length > this.labels.length) {
-                this.labels.push(dataset.data.length);
+            //add datapoint to existing dataset
+            if (data.xValue == null) {
```
Comment (antonykamp) [inline static/scripts/PlotHandler.js:69]: If you want, you can use simply type `if(!data.xValue)`

**Review by antonykamp — CHANGES_REQUESTED** [review]

File: static/scripts/newblocks.js [hunk 2]
```diff
@@ -799,7 +799,7 @@ Blockly.JavaScript["plotting_one_value"] = function (block) {
   var variable_plotName = block.getFieldValue("plotName");
   var variable_plotType = block.getFieldValue("plotType");
 
-  var code = "plot({yValue: ";
+  var code = "plot({xValue: null, yValue: ";
```
[the remark this answers is not in the document]
Comment (kohlros) [reply static/scripts/newblocks.js:802]: I am not sure how it can be avoided.

[the remark this answers is not in the document]
Comment (antonykamp) [reply static/scripts/newblocks.js:802]: Just leave it out :D `data.xValue` should be `null` per default, if you don't declare it

File: static/scripts/PlotHandler.js [hunk 3]
```diff
@@ -73,13 +78,22 @@ class PlotWorker {
...
             const config = {
-                type: this.plotType,
+                //every plot starts as a scatterplot to place the datapoints on the canvas. The actual representation is later specified in each dataset. 
+                type: 'scatter',
                 data: testplotData,
                 options: {
+                    indexAxis: 'x',
+                    scales: {
+                        y: {
+                            beginAtZero: true,
```
[the remark this answers is not in the document]
Comment (kohlros) [reply static/scripts/PlotHandler.js]: It doesn't. However if the x-Axis isn't mentioned in the setup, linegraphs aren't drawn properly and I don't know why. I will take a look into this in the PR where plotting-optics are improved.

**Review by antonykamp — APPROVED** [review]

**Pull request merged**